massa-models/src/endorsement.rs

```diff
@@ -166,7 +168,24 @@ impl SecureShareEndorsement {
 /// Wrapped endorsement
 pub type SecureShareEndorsement = SecureShare<Endorsement, EndorsementId>;
 
-impl SecureShareContent for Endorsement {}
+impl SecureShareContent for Endorsement {
+    fn compute_hash(
+        content: &Self,
+        content_serialized: &[u8],
+        content_creator_pub_key: &PublicKey,
+    ) -> Result<Hash, SerializeError> {
+        let de_data = DenunciationData::Endorsement((content.slot, content.index));
```

Comment:
I find it a bit weird to use a Serializer for DenunciationData because:
* it doesn't really serialize (for example the variant number of the enum is not encoded)
* it can return an Error which complicates everything upstream, but compute_hash does not need to return a Result